cdap-event-reader-spi/src/main/java/io/cdap/cdap/spi/events/StartProgramEvent.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.spi.events;
+
+import javax.annotation.Nullable;
+
+/**
+ * Event from PubSub that signals to start a program.
+ */
+public class StartProgramEvent implements Event<StartProgramEventDetails> {
+
+    private final long publishTime;
+    private final String version;
+    private final StartProgramEventDetails startProgramEventDetails;
+
+    /**
+     * Construct a new StartProgramEvent.
+     *
+     * @param publishTime publish time of event
+     * @param version version of event
+     * @param startProgramEventDetails Details needed to start program
+     */
+    public StartProgramEvent(long publishTime, String version,
+                             StartProgramEventDetails startProgramEventDetails) {
+        this.publishTime = publishTime;
+        this.version = version;
+        this.startProgramEventDetails = startProgramEventDetails;
+    }
+
+    @Override
+    public EventType getType() {
+        return EventType.PROGRAM_START;
+    }
+
+    @Override
+    public long getPublishTime() {
+        return publishTime;
+    }
+
+    @Override
+    public String getVersion() {
+        return version;
+    }
+
+    /**
+     * Get instance name is N/A for incoming events.
```

Comment:
I don't understand this javadoc. What is N/A?

Reply:
If there is no use for incoming events, why have this method declared at the `Event` interface?

Reply:
Outgoing events published information from the CDAP instance to another system. It was helpful to have the instance name in the event so the receiver knew where the event came from. For incoming events, knowing the instance name is not applicable (N/A).

Reply:
Do you think it would be better to have a separate interface for incoming events?

Reply:
It is needed for EventWriter. In the future, it could also be used to pin a job to specific instance. For now, what do you think of the current change: making getInstance default to return null in the Event interface?